src/config.rs

```diff
@@ -428,6 +431,44 @@ impl Default for BookConfig {
             src: PathBuf::from("src"),
             multilingual: false,
             language: Some(String::from("en")),
+            text_direction: None,
+        }
+    }
+}
+
+impl BookConfig {
+    /// Gets the realized text direction, either from [BookConfig::text_direction]
+    /// or derived from [BookConfig::language], to be used by templating engines.
+    pub fn realized_text_direction(&self) -> TextDirection {
+        if let Some(direction) = self.text_direction {
+            direction
+        } else {
+            TextDirection::from_lang_code(&self.language.clone().unwrap_or_default())
```

Comment:
I believe you can use `as_deref` instead and avoid the extra clone here. That way you to from `Option<String>` to `Option<&str>` to `&str` without extre allocations:

```suggestion
            TextDirection::from_lang_code(self.language.as_deref().unwrap_or_default())
```